server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -294,8 +294,8 @@ public ClusterState execute(ClusterState currentState) {
                 }
                 newEntry = new SnapshotsInProgress.Entry(
                         new Snapshot(repositoryName, snapshotId), request.includeGlobalState(), request.partial(),
-                        State.STARTED, indexIds, dataStreams, threadPool.absoluteTimeInMillis(), repositoryData.getGenId(), shards,
-                        null, userMeta, version);
+                        completed(shards.values()) ? State.SUCCESS : State.STARTED, indexIds, dataStreams,
```

Comment:
Also fixes yet another spot where we'd produce a snapshot with all shards in a final state but not a final state for the overall snapshot. I'll try to track down any remaining cases (I think this should've been the last one) if there are any and add an assertion to prevent this kind of bug going forward.